lib/thread/mutex+osx.myr

```diff
@@ -0,0 +1,75 @@
+use std
```

Comment:
As per previous discussion on IRC, we should dedup these things. At the same time, since the code for that just landed, I think I'm ok merging this as is.